packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+  /// string.
+  factory DiagnosticsNode.transformProperty(String name, Matrix4 value, {
+    bool showNull: true,
+  }) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      object: value,
+      header: debugDescribeTransform(value).join('\n'),
+      showNull: showNull,
+    );
+  }
+
+  factory DiagnosticsNode.hasProperty(String name, Object value) {
+    return new _LeafDiagnosticsValue(
+        name: name,
+        object: value,
+        header: value != null ? 'has $name' : '',
+        showNull: false,
+        showName: false
+    );
+  }
+
+
+  /// Property describing an iterable.
+  ///
+  /// If `style` is [DiagnosticsTreeStyle.singleLine], the iterable is described as
+  /// a comma separated list, otherwise the iterable is described as a line
+  /// break separated list.
+  /// Empty values are always hidden as they would otherwise be shown as an
+  /// empty `string` which would be confusing.
+  factory DiagnosticsNode.iterableProperty(String name, Iterable<Object> value, {
+    bool showNull: true,
+    DiagnosticsTreeStyle style: DiagnosticsTreeStyle.singleLine,
+  }) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      object: value,
+      header: style == DiagnosticsTreeStyle.singleLine ?
+        value?.join(', ') : value?.join('\n'),
+      showNull: showNull,
+      style: style,
+      hidden: value != null && value.isEmpty,
+    );
+  }
+
+  /// Diagnostics containing just a string `message` and not a a concrete name
+  /// or value.
+  factory DiagnosticsNode.message(String message, {
+    DiagnosticsTreeStyle style: DiagnosticsTreeStyle.singleLine,
+  }) {
+    return new _LeafDiagnosticsValue(
+      name: '',
+      header: message,
+      style: style,
+    );
+  }
+
+  /// Constructor that creates a [DiagnosticsNode] where properties and children
+  /// are computed lazily.
+  factory DiagnosticsNode.lazy({
+    String name,
+    Object object,
+    String header,
+    void fillProperties(List<DiagnosticsNode> properties),
+    List<DiagnosticsNode> getChildren(),
+    String emptyDescription,
+    DiagnosticsTreeStyle style=DiagnosticsTreeStyle.normal,
+  }) {
+    return new _LazyMembersDiagnosticsNode(
+      name: name,
+      object: object,
+      header: header,
+      fillProperties: fillProperties,
+      getChildren: getChildren,
+      style: style,
+      emptyDescription: emptyDescription,
+    );
+  }
+
+  /// Label describing the Diagnostics node.
+  ///
+  /// * If the DiagnosticsNode represents property, the [name] indicates the
+  ///   name of the property.
+  /// * If the DiagnosticsNode is a child of another DiagnosticsNode, the name
+  ///   describes the relationship between the parent and the child. For
+  ///   example: `child 5`.
+  final String name;
+
+  /// Header with a short text description of the node itself not including
+  /// children or properties.
+  final String header;
+
+  /// Whether to show if the value is null.
```

Comment:
There are definitely cases where null should be shown, e.g. any case where it's not a valid value, or even in general not the default value. Basically any time the value being null is something that is noteworthy. The point of dumping these trees is debugging, and anything unusual, such as a normally non-nullable property being null, is something you want to know about.